apps/teams-test-app/src/components/NestedAppAuthAPIs.tsx

```diff
@@ -12,9 +72,144 @@ const NestedAppAuthAPIs = (): ReactElement => {
+        submit: async (input, setResult) => {
+          try {
+            const targetOrigin = 'https://local.teams.office.com:8080';
+
+            // Check if window.top is accessible
+            if (!window.top) {
+              setResult('Top window not accessible');
+              return 'Top window not accessible';
+            }
+
+            // Define listener for responses
+            const listener = (event: MessageEvent): void => {
+              // Ensure the message comes from the expected origin
+              if (event.origin !== targetOrigin) {
+                console.warn('Received message from an unexpected origin:', event.origin);
+                return;
+              }
+
+              console.log('Received response from top window:', event.data);
+              setResult(JSON.stringify(event.data, null, 2)); // Pretty-print response
+              window.removeEventListener('message', listener);
+            };
+
+            // Add event listener for messages
+            window.addEventListener('message', listener);
+            window.top.postMessage(input, targetOrigin);
+
+            setResult('Message sent to top window, awaiting response...');
+            return 'Message sent to top window';
+          } catch (error) {
+            console.error('Error sending message to top window:', error);
+            setResult(`Error: ${error}`);
+            return `Error: ${error}`;
+          }
+        },
+      },
+      defaultInput: JSON.stringify({
+        id: '2',
+        func: 'nestedAppAuth.execute',
+        args: [],
+        data: NestedAppAuthRequest,
+      }),
+    });
+
+  const AddChildIframeSection = (): React.ReactElement | null => {
+    const [iframeAdded, setIframeAdded] = useState(false);
+
+    const addChildIframe = (): void => {
+      if (iframeAdded) {
+        console.log('Iframe already added.');
+        return;
+      }
+
+      const iframeContainer = document.getElementById('nestedChildiframeContainer');
+      if (!iframeContainer) {
+        console.error('Container not found: nestedChildiframeContainer');
+        return;
+      }
+
+      const childIframe = document.createElement('iframe');
+      childIframe.src = `${window.location.href}?appInitializationTest=true&groupedMode=NestedAppAuthAPIs`;
+      childIframe.id = 'nestedAuthChildIframe';
+      childIframe.width = '100%';
+      childIframe.height = '400px';
+      childIframe.style.border = 'none';
+
+      iframeContainer.appendChild(childIframe);
+      setIframeAdded(true);
+    };
+
+    return (
+      <div style={{ border: '5px solid black', padding: '2px', margin: '2px' }}>
+        <h2>Add Nested Child Iframe</h2>
+        <input
+          name="button_addNestedChildIframe"
+          type="button"
+          value="Add Child Iframe"
+          onClick={addChildIframe}
+          disabled={iframeAdded}
+        />
+        <div
+          id="nestedChildiframeContainer"
```

Comment:
If we're being sticklers about camel casing this should instead be 
```suggestion
          id="nestedChildIframeContainer"
```

(don't forget to update places that reference this id directly)

Reply:
My bad. Thanks. Also, changes at all the relevant places.